pkg/indexerlookup/candidatefinder_test.go

```diff
@@ -0,0 +1,147 @@
+package indexerlookup_test
+
+import (
+	"context"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/filecoin-project/lassie/pkg/indexerlookup"
+	"github.com/filecoin-project/lassie/pkg/internal/mockindexer"
+	"github.com/filecoin-project/lassie/pkg/internal/testutil"
+	"github.com/filecoin-project/lassie/pkg/types"
+	"github.com/ipfs/go-cid"
+	"github.com/ipni/storetheindex/api/v0/finder/model"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCandidateFinder(t *testing.T) {
+	cids := make([]cid.Cid, 0, 10)
+	candidates := make(map[cid.Cid][]types.RetrievalCandidate, 10)
+	binaryMetadata := make(map[cid.Cid][][]byte, 10)
+	for i := 0; i < 10; i++ {
+		next := testutil.GenerateRetrievalCandidates(t, 2)
```

Comment:
```suggestion
		next := testutil.GenerateRetrievalCandidates(t, 1)
```
am I missing something?

Reply:
yea we use both candidates

Reply:
see the test cases